### ISSUE TITLE
[Feature Request] Kotlin/JS support

### ISSUE THREAD

**Issue by Jolanrensen** [GoogleChromeLabs/comlink#429]
Hi, 
I'm just gonna put this here.
I'm not sure if it's even possible (my brain broke trying to convert / adapt comlink to work with Kotlin JS), but it would be awesome if it could work!

Kotlin is great for multihtreading using coroutines, however, multithreading in JavaScript is limited to workers. A library like comlink or something comparable would be a really powerful tool to enhance Kotlin JS. I'm especially intrigued by the idea of accessing objects from multiple threads, just like in for instance the JVM.
If combined with inline workers (like how https://github.com/developit/greenlet does it) and constrained by interfaces, I think something like Comlink could really boost the adaption of Kotlin JS.

**Comment by jccr** [comment]
I've been prototyping this idea for an internal project of mine, so to a degree I know it's possible.

**Comment by Jolanrensen** [comment]
Right, yeah that's different. Probably easier, as the JVM already offers threading. However, while you can do some Javafx stuff in browsers, I think JavaScript is still faster and better suited to modern day browsers (correct me if I'm wrong). It just needs, imo, a bit of kotlin and multithreading to be more enjoyable and useful.
